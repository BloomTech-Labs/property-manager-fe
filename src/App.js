import React from 'react';
import { Router } from '@reach/router';
import { BreakpointProvider, setDefaultBreakpoints } from 'react-socks';
// Landing imports
import LandingPage from './views/landing/LandingPage';
import Landlord from './views/landing/Landlord';
import Tenant from './views/landing/Tenant';
import Features from './views/landing/Features';
import Contact from './views/landing/Contact';
import Home from './views/landing/Home';
// import GetStarted from './views/SignUp';
import ProtectedRoutes from './components/Auth/Routes/ProtectedRoutes';

setDefaultBreakpoints([{ mobile: 250 }, { tablet: 769 }, { desktop: 1025 }]);

const App = () => {
  return (
    <BreakpointProvider>
      <div className="App">
        <Router>
          <LandingPage path="/">
            <Home path="/" />
            <Landlord path="landlord" />
            <Tenant path="tenant" />
            <Features path="features" />
            <Contact path="contact" />
          </LandingPage>
          <ProtectedRoutes path="dashboard/*" />
        </Router>
      </div>
    </BreakpointProvider>
  );
};

export default App;
