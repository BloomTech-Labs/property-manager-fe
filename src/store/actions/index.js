// IMPORTS/INITIALIZATION =========================|
// ================================================|
import axios from 'axios';
import axiosAuth from '../../helpers/axiosAuth';
// ------------------------------------------------|
// AUTH ACTIONS ===================================|
// ================================================|
export const AUTH_REQUEST_START = 'AUTH_REQUEST_START';
export const AUTH_REQUEST_SUCCESS = 'AUTH_REQUEST_SUCCESS';
export const AUTH_REQUEST_FAIL = 'AUTH_REQUEST_FAIL';
// ------------------------------------------------|
// LOGIN / SIGNUP ---------------------------------|
export const auth = url => (email, password) => async dispatch => {
  dispatch({ type: AUTH_REQUEST_START });

  try {
    const res = await axios.post(url, {
      email,
      password
    });

    localStorage.setItem('token', res.data.token);

    dispatch({ type: AUTH_REQUEST_SUCCESS, payload: { token: res } });
  } catch (err) {
    console.error(err);
    dispatch({ type: AUTH_REQUEST_FAIL, payload: { errorMessage: err } });
  }
};
// ------------------------------------------------|
// PROPERTY ACTIONS ===============================|
// ================================================|
export const ADD_PROPERTY_START = 'ADD_PROPERTY_START';
export const ADD_PROPERTY_SUCCESS = 'ADD_PROPERTY_SUCCESS';
export const ADD_PROPERTY_FAIL = 'ADD_PROPERTY_FAIL';
export const GET_PROPERTY_START = 'GET_PROPERTY_START';
export const GET_PROPERTY_SUCCESS = 'GET_PROPERTY_SUCCESS';
export const GET_PROPERTY_FAIL = 'GET_PROPERTY_FAIL';
// ------------------------------------------------|
// CREATE A PROPERTY ------------------------------|
export const createProperty = url => property => async dispatch => {
  dispatch({ type: ADD_PROPERTY_START });
  console.log(property);
  try {
    const res = await axios.post(url, { ...property });

    dispatch({
      type: ADD_PROPERTY_SUCCESS,
      payload: {
        created: res
      }
    });
  } catch (err) {
    console.error(err);

    dispatch({ type: ADD_PROPERTY_FAIL, payload: { errorMessage: err } });
  }
};
// ------------------------------------------------|
// GET PROPERTIES ---------------------------------|
export const getProperties = url => async dispatch => {
  dispatch({ type: GET_PROPERTY_START });

  try {
    const res = await axiosAuth().get(url);

    dispatch({
      type: GET_PROPERTY_SUCCESS,
      payload: {
        properties: res.data
      }
    });
  } catch (err) {
    console.error(err);

    dispatch({ type: GET_PROPERTY_FAIL, payload: { errMsg: err } });
  }
};
// ------------------------------------------------|
// USER ACTIONS ===================================|
// ================================================|

// define your user actions here

// ------------------------------------------------|
