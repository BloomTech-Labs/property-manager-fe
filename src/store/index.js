import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { getFirebase } from 'react-redux-firebase';
import reducer from './reducers';

const middleware =
  process.ENV === 'production'
    ? [thunk.withExtraArgument(getFirebase)]
    : [thunk.withExtraArgument(getFirebase), logger];

export const makeStore = state =>
  createStore(
    reducer,
    state,
    compose(
      applyMiddleware(...middleware),

      /* eslint-disable */
      window.__REDUX_DEVTOOLS_EXTENSION__
        ? window.__REDUX_DEVTOOLS_EXTENSION__()
        : f => f
    )
  );

export default makeStore();
