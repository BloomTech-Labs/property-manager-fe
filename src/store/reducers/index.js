import { combineReducers } from 'redux';
import authReducer from './authReducer';
import propReducer from './propReducer';
import getUserReducer from './getUserReducer';
import workOrderReducer from './workOrderReducer';

export default combineReducers({
  authReducer,
  propReducer,
  getUserReducer,
  workOrderReducer
});
