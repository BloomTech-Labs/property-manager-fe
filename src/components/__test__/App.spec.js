import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../store';
import App from '../../App';

afterEach(rtl.cleanup);

test('Render easy communication', async () => {
  // render our React app into an in-memory DOM so we can test against it
  const wrapper = await rtl.render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(await wrapper.queryAllByText(/easy communication/i));
});

test('snapshot', async () => {
  const { container } = await rtl.render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(container.firstChild).toMatchSnapshot();
});
