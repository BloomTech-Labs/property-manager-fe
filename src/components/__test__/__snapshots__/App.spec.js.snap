// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <nav
    class="nav-top"
  >
    <a
      aria-current="page"
      class="logo-link"
      href="/"
    >
      <img
        alt="Freehold logo, link home"
        class="nav-logo"
        data-testid="nav-logo"
        src="logo.png"
      />
    </a>
    <ul>
      <li>
        <a
          alt="Features"
          class="nav-link"
          href="/#spotlight"
        >
          Features
        </a>
      </li>
      <li>
        <a
          alt="Pricing"
          class="nav-link"
          href="/#pricing"
        >
          Pricing
        </a>
      </li>
      <a
        class="nav-link"
        href="/login"
      >
        Login
      </a>
    </ul>
  </nav>
  <div
    style="outline: none;"
    tabindex="-1"
  >
    <div
      class="landing-container"
    >
      <div
        class="landing-content"
      >
        <div
          class="header"
        >
          <div
            class="left"
          >
            <img
              alt="housing"
              class="header-image"
              src="header-image.svg"
            />
            <h4
              class="left-text"
            >
              Property Management doesn't have to be hard.
            </h4>
          </div>
          <div
            class="right"
          >
            <div>
              <h4>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-comments fa-w-18 icon"
                  data-icon="comments"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 192c0-88.4-93.1-160-208-160S0 103.6 0 192c0 34.3 14.1 65.9 38 92-13.4 30.2-35.5 54.2-35.8 54.5-2.2 2.3-2.8 5.7-1.5 8.7S4.8 352 8 352c36.6 0 66.9-12.3 88.7-25 32.2 15.7 70.3 25 111.3 25 114.9 0 208-71.6 208-160zm122 220c23.9-26 38-57.7 38-92 0-66.9-53.5-124.2-129.3-148.1.9 6.6 1.3 13.3 1.3 20.1 0 105.9-107.7 192-240 192-10.8 0-21.3-.8-31.7-1.9C207.8 439.6 281.8 480 368 480c41 0 79.1-9.2 111.3-25 21.8 12.7 52.1 25 88.7 25 3.2 0 6.1-1.9 7.3-4.8 1.3-2.9.7-6.3-1.5-8.7-.3-.3-22.4-24.2-35.8-54.5z"
                    fill="currentColor"
                  />
                </svg>
                 Easy Communication
              </h4>
              <br />
              <h4>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-tasks fa-w-16 icon"
                  data-icon="tasks"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M139.61 35.5a12 12 0 0 0-17 0L58.93 98.81l-22.7-22.12a12 12 0 0 0-17 0L3.53 92.41a12 12 0 0 0 0 17l47.59 47.4a12.78 12.78 0 0 0 17.61 0l15.59-15.62L156.52 69a12.09 12.09 0 0 0 .09-17zm0 159.19a12 12 0 0 0-17 0l-63.68 63.72-22.7-22.1a12 12 0 0 0-17 0L3.53 252a12 12 0 0 0 0 17L51 316.5a12.77 12.77 0 0 0 17.6 0l15.7-15.69 72.2-72.22a12 12 0 0 0 .09-16.9zM64 368c-26.49 0-48.59 21.5-48.59 48S37.53 464 64 464a48 48 0 0 0 0-96zm432 16H208a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h288a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H208a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h288a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H208a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h288a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
                    fill="currentColor"
                  />
                </svg>
                 Easy Management
              </h4>
              <br />
              <h4>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-spa fa-w-18 icon"
                  data-icon="spa"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M568.25 192c-29.04.13-135.01 6.16-213.84 83-33.12 29.63-53.36 63.3-66.41 94.86-13.05-31.56-33.29-65.23-66.41-94.86-78.83-76.84-184.8-82.87-213.84-83-4.41-.02-7.79 3.4-7.75 7.82.23 27.92 7.14 126.14 88.77 199.3C172.79 480.94 256 480 288 480s115.19.95 199.23-80.88c81.64-73.17 88.54-171.38 88.77-199.3.04-4.42-3.34-7.84-7.75-7.82zM287.98 302.6c12.82-18.85 27.6-35.78 44.09-50.52 19.09-18.61 39.58-33.3 60.26-45.18-16.44-70.5-51.72-133.05-96.73-172.22-4.11-3.58-11.02-3.58-15.14 0-44.99 39.14-80.27 101.63-96.74 172.07 20.37 11.7 40.5 26.14 59.22 44.39a282.768 282.768 0 0 1 45.04 51.46z"
                    fill="currentColor"
                  />
                </svg>
                 Peace of mind
              </h4>
              <button
                class="header-button"
                type="button"
              >
                Get Started
              </button>
            </div>
          </div>
        </div>
        <div
          class="spotlightWrapper"
          id="spotlight"
        >
          <div
            class="spotHeadingWrapper"
          >
            <img
              alt="Orange computer and phone"
              class="spotHeaderImage"
              src="landing-3homes-middle.svg"
            />
            <h1
              class="spotHeading"
            >
              Manage your properties, don’t let them manage you!
            </h1>
          </div>
          <div
            class="spotlightCards"
          >
            <div
              class="spotlightCard"
            >
              <img
                alt="Spotlight card with documents."
                class="spotCardImages"
                src="landing-docu-card.svg"
              />
              <hr
                class="spotDivider"
              />
              <h5
                class="spotCardContent"
              >
                Communicate and share documents with you tenants directly in the app.
              </h5>
            </div>
            <div
              class="spotlightCard"
            >
              <img
                alt="Spotlight card highlghting Stripe."
                class="spotCardImages"
                src="landing-stripe.svg"
              />
              <hr
                class="spotDivider"
              />
              <h5
                class="spotCardContent"
              >
                Accept payments with confidence through Stripe’s secure API.
              </h5>
            </div>
            <div
              class="spotlightCard"
            >
              <img
                alt="Spotlight card highlighting work orders."
                class="spotCardImages"
                src="landing-workorder.svg"
              />
              <hr
                class="spotDivider"
              />
              <h5
                class="spotCardContent"
              >
                Organize and review repair requests from creation to completion.
              </h5>
            </div>
            <div
              class="spotlightCard"
            >
              <img
                alt="Spotlight card highlighting statistics and charts."
                class="spotCardImages"
                src="landing-chart.svg"
              />
              <hr
                class="spotDivider"
              />
              <h5
                class="spotCardContent"
              >
                Keep track of your cash flow month-to-month so you know exactly where your finances are.
              </h5>
            </div>
          </div>
          <div
            class="btnWrapper"
          >
            <button
              class="spotBtn"
              type="submit"
            >
              Sign up
            </button>
          </div>
        </div>
        <div
          id="pricing"
        >
          <div
            class="featureHeadingWrapper"
          >
            <img
              alt="Orange computer and phone"
              class="featureImage"
              src="features-image.svg"
            />
            <h1
              class="featureHeading"
            >
              Which Freehold is right for me?
            </h1>
          </div>
          <div
            class="featureTableWrapper"
          >
            <table
              class="featureTable"
            >
              <thead
                class="featureTiersWrapper"
              >
                <tr
                  class="featureTiers"
                >
                  <th />
                  <th>
                    Free
                  </th>
                  <th>
                    $49/mo
                  </th>
                  <th>
                    $99/mo
                  </th>
                </tr>
              </thead>
              <tbody
                class="featureTableBodyWrapper"
              >
                <tr
                  class="tableRow"
                >
                  <td
                    class="featureName"
                  >
                    Professional dashboard
                  </td>
                  <td>
                    ✔
                  </td>
                  <td>
                    ✔
                  </td>
                  <td>
                    ✔
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="featureName"
                  >
                    Emailing service
                  </td>
                  <td>
                    ✔
                  </td>
                  <td>
                    ✔
                  </td>
                  <td>
                    ✔
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="featureName"
                  >
                    Workorder management
                  </td>
                  <td>
                    ✔
                  </td>
                  <td>
                    ✔
                  </td>
                  <td>
                    ✔
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="featureName"
                  >
                    Accept payments
                  </td>
                  <td />
                  <td>
                    ✔
                  </td>
                  <td>
                    ✔
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="featureName"
                  >
                    Store documents
                  </td>
                  <td />
                  <td>
                    ✔
                  </td>
                  <td>
                    ✔
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="featureName"
                  >
                    Downloadable applications
                  </td>
                  <td />
                  <td>
                    ✔
                  </td>
                  <td>
                    ✔
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="featureName"
                  >
                    Premium messaging service
                  </td>
                  <td />
                  <td />
                  <td>
                    ✔
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="featureName"
                  >
                    24/7 Premium support
                  </td>
                  <td />
                  <td />
                  <td>
                    ✔
                  </td>
                </tr>
                <tr
                  class="tableRow"
                >
                  <td
                    class="featureName"
                  >
                    Itemized expense tracking
                  </td>
                  <td />
                  <td />
                  <td>
                    ✔
                  </td>
                </tr>
              </tbody>
            </table>
            <div
              class="buy-btn-wrapper"
            >
              <div
                class="empty"
              />
              <button
                class="buy-btn"
                type="submit"
              >
                Buy
              </button>
              <button
                class="buy-btn"
                type="submit"
              >
                Buy
              </button>
              <button
                class="buy-btn"
                type="submit"
              >
                Buy
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="footerWrapper"
  >
    <div
      class="footerCopyWrapper"
    >
      <div
        class="footerLogo"
      >
        <a
          aria-current="page"
          href="/"
        >
          <img
            alt="Freehold logo"
            src="footerLogo.png"
          />
        </a>
        <h3
          class="footerCopyright"
        >
          © Copyright Freehold 
          2020
        </h3>
      </div>
    </div>
    <a
      class="footerLink"
      href="/contact"
    >
      Contact Us
    </a>
  </div>
</div>
`;
